framework/editor/TextEditor.kt

```diff
@@ -143,7 +146,7 @@ object TextEditor {
     private fun createState(
         bottomSpace: Dp,
         processorFn: (
-            content: SnapshotStateList<AnnotatedString>,
+            content: SnapshotStateList<GlyphLine>,
```

Comment:
On review, there are actually very few places this is going to affect, we already have largely implemented it. The `highlight` function should actually take a function that returns an AnnotatedString rather than a GlyphLine, so that is the correct generalised interface.

Basically we should just double-check the implementations, but it seems largely correct already